parasol/solr/client.py

```diff
@@ -0,0 +1,131 @@
+import inspect
+import json
+import logging
+import sys
+import time
+from urllib.parse import urljoin
+
+from attrdict import AttrDict
+import requests
+
+# Start making Django agnostic from the get-go
+try:
+    from django.settings import settings
+    from django.core.exceptions import ImproperlyConfigured
+except ImportError:
+    pass
+
+
+from parasol import __version__ as parasol_version
+from parasol.solr.schema import Schema
+from parasol.solr.update import Update
+from parasol.solr.admin import CoreAdmin
+
+logger = logging.getLogger(__name__)
+
+
+
+class SolrClientException(Exception):
+    '''Base class for all exceptions in this module'''
+    pass
+
+
+class CoreForTestExists(SolrClientException):
+    '''Raised when default core for running unit tests exists'''
+
+
+
+class SolrClient:
+    '''Base class for all SolrClient with sane development defaults'''
+    #: Url for solr base instance
+    solr_url = 'http://localhost:8983/solr'
+    #: core admin handler
+    core_admin_handler = 'admin/cores'
+    #: select handler
+    select_handler = 'select'
+    #: schema handler
+    schema_handler = 'schema'
+    # update handler
+    update_handler = 'update'
+    #: core or collection
+    collection = ''
+    # commitWithin definition
+    commit_within = 1000
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Named args would be cleaner and allow for ordered input as well (i.e. `SolrClient('myserver:8983/solr', 'mycore')`.
It also seems not useful and possibly bad to store arbitrary keyword args we don't do anything with.

I see you said "for now" in the comments, so maybe this was already your plan.

Reply:
That was roughly already my plan. I tend to write a flexible interface (too flexible) to start with, and then tighten it as a good pattern for usage emerges.